cmd/thanos/receive.go

```diff
@@ -408,25 +344,301 @@ func runReceive(
 	}
 
 	{
-		if limiter.CanReload() {
-			ctx, cancel := context.WithCancel(context.Background())
-			g.Add(func() error {
-				level.Debug(logger).Log("msg", "limits config initialized with file watcher.")
-				if err := limiter.StartConfigReloader(ctx); err != nil {
-					return err
-				}
-				<-ctx.Done()
-				return nil
-			}, func(err error) {
-				cancel()
-			})
+		// Config reloader.
+		if err := configReloader(limiter, logger, g); err != nil {
+			return err
+		}
+
+	}
+	level.Info(logger).Log("msg", "starting receiver")
+	return nil
+}
+
+func routerMode(
+	conf *receiveConfig,
+	logger log.Logger,
+	reg *prometheus.Registry,
+	comp component.SourceStoreAPI,
+	dbs *receive.MultiTSDB,
+	writer *receive.Writer,
+	g *run.Group,
+	hashringChangedChan chan struct{},
+	statusProber prober.Probe,
+	relabelConfig []*relabel.Config,
+	receiveMode receive.ReceiverMode,
+	tracer opentracing.Tracer,
+	dialOpts []grpc.DialOption,
+	limiter *receive.Limiter,
+	httpProbe *prober.HTTPProbe,
+	grpcProbe *prober.GRPCProbe,
+	grpcLogOpts []grpc_logging.Option,
+	tagOpts []tags.Option,
+
+) error {
+
+	err, webHandler := getWebhandler(conf, logger, reg, dbs, writer, relabelConfig, receiveMode, tracer, dialOpts, limiter)
+	if err != nil {
+		return err
+	}
+
+	enableIngestion := false
+	level.Debug(logger).Log("msg", "setting up hashring")
+	{
+		if err := setupHashring(g, logger, reg, conf, hashringChangedChan, webHandler, statusProber, enableIngestion); err != nil {
+			return err
+		}
+	}
+
+	// Setting up HTTP server.
+	level.Debug(logger).Log("msg", "setting up HTTP server")
+	{
+		if err := setupHttpSever(logger, reg, comp, httpProbe, conf, g, statusProber); err != nil {
+			return err
+		}
+	}
+
+	// Setting up gRPC.
+	level.Debug(logger).Log("msg", "setting up gRPC server")
+	{
+		if err := setupgRPC(logger, conf, webHandler, reg, tracer, grpcLogOpts, tagOpts, comp, grpcProbe, statusProber, g, dbs, httpProbe); err != nil {
+			return nil
+		}
+	}
+
+	// Setting up HTTP handler.
+	level.Debug(logger).Log("msg", "setting up receive HTTP handler")
+	{
+		setupHttpHandler(g, webHandler)
+	}
+
+	{
+		// Config reloader.
+		if err := configReloader(limiter, logger, g); err != nil {
```

Comment:
The comment is essentially the name the of the function which is not providing any relevant information. If you have anything informative about that function or the logic then only put a comment.

Reply:
Alright, thanks for the review